lib/SetAPI/generic/SetInterfaceV1.py

```diff
@@ -103,43 +215,100 @@ def get_set(
 
         :param self: this class
         :type self: SetInterfaceV1
-        :param ref: object reference
-        :type ref: str
-        :param include_item_info: whether or not to populate the info of the set items, defaults to False
-        :type include_item_info: bool, optional
-        :param ref_path_to_set: ref path for the object, defaults to None
-        :type ref_path_to_set: list[str] | None, optional
-        :param include_set_item_ref_paths: whether or not to add ref paths, defaults to False
-        :type include_set_item_ref_paths: bool, optional
+        :param params: dictionary of parameters for the request
+        :type params: dict[str, Any]
         :return: set data
         :rtype: dict[str, Any]
         """
-        if ref_path_to_set is None:
-            ref_path_to_set = []
-        obj_selector = build_ws_obj_selector(ref, ref_path_to_set)
+        checked_params = self._check_get_set_params(params)
 
-        # retrieve the object from the workspace
-        ws_data = self.ws.get_objects2({"objects": [obj_selector]})
+        obj_selector = build_ws_obj_selector(
+            checked_params["ref"], checked_params[REF_PATH_TO_SET]
+        )
 
+        resp = self.ws_client.get_objects2({"objects": [obj_selector]})
         set_data = {
-            "data": ws_data["data"][0]["data"],
-            "info": ws_data["data"][0]["info"],
+            "data": resp["data"][0]["data"],
+            "info": resp["data"][0]["info"],
         }
+        kbase_set_type = set_data["info"][2]
+
+        if "KBaseSets" not in kbase_set_type:
```

Comment:
Up to this point, all that this method has done is to retrieve an object (with the ref supplied in the params) from the workspace. It has no idea whether the object is a set, a set constituent, a KBaseNarrative.Narrative object, etc. Now we check the type of the fetched object to see if it's a set of some sort and if so, populate the members.